File: libs/language/src/lib/mapping/__tests__/schema/personValueExample.ts
import { z } from 'zod';
import {
  blueDescriptionField,
  blueNameField,
  withTypeBlueId,
  blueNodeField,
} from '../../annotations';

export const personValueExampleSchema = withTypeBlueId('PersonValue-BlueId')(
  z.object({
    age1: z.number().optional(),
    age2Name: blueNameField('age2'),
    age2Description: blueDescriptionField('age2'),
    age2: z.number().optional(),
    age3Name: blueNameField('age3'),
    age3: blueNodeField().optional(),
  })
);

export type PersonValueExample = z.infer<typeof personValueExampleSchema>;

File: references/java-sdk/src/test/java/blue/language/mapping/model/PersonValueExample.java
package blue.language.mapping.model;

import blue.language.model.*;

@TypeBlueId("PersonValue-BlueId")
public class PersonValueExample {
    public Integer age1;
    @BlueName("age2")
    public String age2Name;
    @BlueDescription("age2")
    public String age2Description;
    public Integer age2;
    public Node age3;

    @Override
    public String toString() {
        return "PersonValueExample{" +
               "age1=" + age1 +
               ", age2Name='" + age2Name + '\'' +
               ", age2Description='" + age2Description + '\'' +
               ", age2=" + age2 +
               ", age3=" + age3 +
               '}';
    }
}